### PR TITLE
[lldb][test] Add test-coverage for DW_AT_APPLE_objc_complete_type parsing

### PR DESCRIPTION
**Pull request by Michael137**
When given a DIE for an Objective-C interface (which doesn't have a `DW_AT_APPLE_objc_complete_type`), the `DWARFASTParserClang` will try to find the DIE which corresponds to the implementation to complete the interface DIE. The code is here:
https://github.com/llvm/llvm-project/blob/d2e7ee77d33e8b3be3b1d4e9bc5bc4c60b62b554/lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp#L1718-L1738

However, this was currently not exercised in our test-suite (removing the code above didn't fail any LLDB test).

This patch adds a test which exercises this codepath (it will fail if we don't fetch the implementation DIE in the `DWARFASTParserClang`).

Something that's not currently clear to me is why `frame var *f` succeeds even without the `DW_AT_APPLE_objc_complete_type` infrastructure. If it's using the ObjC runtime, we should make `expr` do the same, in which case we can remove this code from `DWARFASTParserClang`.

### CODE REVIEW

**Comment by Michael137** [comment]
[`TestHiddenIvars.py`](https://github.com/llvm/llvm-project/blob/d2e7ee77d33e8b3be3b1d4e9bc5bc4c60b62b554/lldb/test/API/lang/objc/hidden-ivars/TestHiddenIvars.py) looks like it intended to test the same codepath, but it passes cleanly with and without the calls to `FindCompleteObjCDefinitionTypeForDIE`

File: lldb/test/Shell/Expr/TestObjCHiddenIvars.test [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Tests that LLDB correctly finds the implementation
+# DIE (with a `DW_AT_APPLE_objc_complete_type`)
+# given an interface DIE (without said attribute).
+#
+# RUN: split-file %s %t
+# RUN: %clangxx_host %t/lib.m  -c -g -o %t/lib.o
+# RUN: %clangxx_host %t/main.m -c -g -o %t/main.o
+# RUN: %clangxx_host %t/main.o %t/lib.o -o %t/a.out -framework Foundation
```
Comment (labath) [inline lldb/test/Shell/Expr/TestObjCHiddenIvars.test:8]: I think this needs a `REQUIRES: system-darwin`.

For science, I also tried to make a version of this test which could run on other systems, but all I managed is to hit an (lldb)assert:
```
$ cat a.m 
__attribute__((objc_root_class)) @interface  Foo  {
int y;
}
@end

#ifdef DEF
@implementation Foo {
    int i;
}

- (id)init {
  self->i = 1;
  self->y = 2;

  return self;
}
@end
#endif

Foo *foo;
$ clang -c a.m -g -DDEF --target=aarch64-apple-macosx
$ lldb a.o -o "target variable foo[0]"
(lldb) target create "a.o"
Current executable set to '/tmp/a.o' (arm64).
(lldb) target variable foo[0]
warning: trying to determine the size of type @interface Foo{
    int y;
    int i;
}
@end
without a valid ExecutionContext. this is not reliable. please file a bug against LLDB.
backtrace:
 #0 0x00007f117b0ab29e llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) llvm/lib/Support/Unix/Signals.inc:723:22
 #1 0x00007f117af11e90 lldb_private::TypeSystemClang::GetObjCBitSize(clang::QualType, lldb_private::ExecutionContextScope*) lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp:4780:17
```

**Review by labath — APPROVED** [review]
> Something that's not currently clear to me is why frame var *f succeeds even without the DW_AT_APPLE_objc_complete_type infrastructure. If it's using the ObjC runtime, which should make expr do the same, in which case we can remove this code from DWARFASTParserClang.

I don't know, but I'm all for deleting code :D

File: lldb/test/Shell/Expr/TestObjCHiddenIvars.test (hunk 1 above)
Comment (Michael137) [reply lldb/test/Shell/Expr/TestObjCHiddenIvars.test:8]: Thanks for trying! Yea I'll probably add the `Darwin` requirement then

**Pull request merged**